### ISSUE TITLE
Feature Proposal: `releaseAfterExceptionBackoff()`

### ISSUE THREAD

**Issue by tylerssn** [spatie/laravel-rate-limited-job-middleware#20]
`releaseAfterBackoff()` when based on the attempted value only can be overkill in certain situations. For example, if I have a job configured for `->allow(2)->everySecond()->releaseAfterSecond()` Laravel will increment the `->attempts()` value, though the job was never technically tried - it was just put on hold because there were already 2 jobs running. In this scenario, you may only want to apply an exponential backoff when an exception is thrown.

Where this code will exponentially back-off after each attempt.

```
(new RateLimited())
            ->allow(2)
            ->everySecond()
            ->releaseAfterSeconds(2)
            ->releaseAfterBackoff($this->attempts(), $backoffRate);
```

This code, and proposed feature, will only backoff based on the exception count. Otherwise, the standard `releaseAfterSeconds()` will be used.

```
(new RateLimited())
            ->allow(2)
            ->everySecond()
            ->releaseAfterSeconds(2)
            ->releaseAfterExceptionBackoff($backoffRate);
```

If this is a feature that would be considered, I'll try to figure out if the current exception count is available to the job and create the PR.

**Comment by tylerssn** [comment]
Assuming `$this->cache->increment('job-exceptions:'.$uuid)` in `Illuminate\Queue\Worker::markJobAsFailedIfWillExceedMaxExceptions()` refers to the global default cache, then `releaseAfterExceptionBackoff()` could use: `(int) Cache::get('job-exceptions:'.$uuid)`.

If it is not worth adding the additional `releaseAfterExceptionBackoff()` method, then the documentation can be updated with an example of how to use this with the existing method:

```php
/**
 * Only after exceptions are logged will the backoff rate be applied.
 * 
 * Attempt 1: Release after 5 seconds
 * Attempt 2: Release after 5 seconds
 * Attempt ...: Release after 5 seconds
 * Exception 1: Release after 5 seconds
 * Exception 2: Release after 15 seconds
 * Exception 3: Release after 35 seconds
 * Exception 4: Release after 75 seconds
 */
public function middleware()
{
    $rateLimitedMiddleware = (new RateLimited())
        ->allow(30)
        ->everySeconds(60)
        ->releaseAfterSeconds(5)
        ->releaseAfterBackoff(Cache::get('job-exceptions:'.$uuid));

    return [$rateLimitedMiddleware];
}
```

**Comment by freekmurze** [comment]
I'd accept a PR to update the docs. Thanks 👍
